shell/platform/windows/win32_window.cc

```diff
@@ -158,16 +158,59 @@ Win32Window::MessageHandler(HWND hwnd,
         tracking_mouse_leave_ = false;
         break;
       case WM_LBUTTONDOWN:
+        // Capture the pointer in case the user drags outside the client area.
+        // In this case, the "mouse leave" event is delayed until the user
+        // releases the button.
+        SetCapture(hwnd);
         xPos = GET_X_LPARAM(lparam);
         yPos = GET_Y_LPARAM(lparam);
         window->OnPointerDown(static_cast<double>(xPos),
-                              static_cast<double>(yPos));
+                              static_cast<double>(yPos), WM_LBUTTONDOWN);
         break;
       case WM_LBUTTONUP:
         xPos = GET_X_LPARAM(lparam);
         yPos = GET_Y_LPARAM(lparam);
         window->OnPointerUp(static_cast<double>(xPos),
-                            static_cast<double>(yPos));
+                            static_cast<double>(yPos), WM_LBUTTONUP);
+        ReleaseCapture();
+        break;
+      case WM_RBUTTONDOWN:
```

Comment:
I thought the most common scenario would be dragging out of the window with the left click. It would seem weird to me to be able to drag with other buttons as well, but we could definitely support that.